src/panoramic/cli/connections.py

```diff
@@ -0,0 +1,281 @@
+from typing import Any, Dict, Optional, Tuple
+
+import click
+import hologram
+import yaml
+
+from panoramic.cli.config.storage import read_config, update_config
+from panoramic.cli.errors import ConnectionNotFound
+from panoramic.cli.paths import Paths
+from panoramic.cli.print import echo_info
+
+CONNECTION_KEYS = [
+    'type',
+    'user',
+    'host',
+    'port',
+    'password',
+    'database',
+    'schema',
+    'warehouse',
+    'account',
+    'project',
+    'key_file',
+]
+
+
+def create_connection_command(
+    name,
+    type,
+    user,
+    host,
+    port,
+    password,
+    password_stdin,
+    database,
+    schema,
+    warehouse,
+    account,
+    project,
+    key_file,
+    no_test,
+):
+    """CLI command. Create new connection."""
+    connections = Connections.load()
+    if name in connections:
+        raise click.ClickException(f'Connection with name "{name}" already exists.')
```

Comment:
I'd probably prefer if we had special classes for these errors. What do you think @raethlo ?

Reply:
I agree that it'd make it more consistent to how we handle it in other parts of the codebase.

Reply:
Added custom errors.